cilium/cmd/bpf_sha_list.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/command"
+	"github.com/cilium/cilium/pkg/defaults"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	shaTitle       = "Datapath SHA"
+	endpointsTitle = "Endpoint(s)"
+
+	noSHA     = "<No template used>"
+	brokenSHA = "<Template path invalid>"
+)
+
+var (
+	stateDir     = filepath.Join(defaults.RuntimePath, defaults.StateDir)
```

Comment:
`RunDir` may not always be this value. maybe provide an option to override it if necessary in case the user has overriden it in the daemon's configuration? This use case isn't common, but it is possible.

Reply:
This isn't the only tool with this problem, we'd need to fix it up for multiple cli commands. That said it's such an unusual case that I don't currently intend on working on this.